src/services/balancer/pools/weighted-pool.service.ts

```diff
@@ -75,8 +75,23 @@ export default class WeightedPoolService {
     createPoolTransaction: TransactionResponse
   ): Promise<CreatePoolReturn> {
     const receipt: any = await createPoolTransaction.wait();
-    const events = receipt.events.filter(e => e.event === 'PoolCreated');
-    const poolAddress = events[0].args[0];
+    let poolAddress;
+    if (receipt.events) {
+      const events = receipt.events.filter(e => e.event === 'PoolCreated');
+      if (events.length > 0 && events[0].args.length > 0) {
+        poolAddress = events[0].args[0];
+      }
+    }
+
+    if (!poolAddress) {
+      const logs = receipt.logs.filter(
+        l =>
+          l.topics?.length > 0 &&
+          l.topics[0] ===
+            '0xa9ba3ffe0b6c366b81232caab38605a0699ad5398d6cce76f91ee809e322dafc' // PoolCreated topic
```

Comment:
This topic address is the same on all networks

Reply:
Yeah so you could put it into a topics.ts constants file I guess so that you can access like `TOPICS.PoolCreated`